### DISCUSSION TITLE
How do we use tab completion?

### DISCUSSION THREAD

**Discussion by leftyfb** [mikefarah/yq#1530] [Q&A] [answered]
No documentation or examples on how to use tab completion. I see that is works to complete the 5 "Available Commands" and that works just fine. But the bash completion is 1000 lines of code which looks like it should be able to parse the fields from a yaml file. None of the incantations of yq seems to complete anything other than the 5 commands. The ordering of the filename being after the field search is also not conducive to how tab completion would work. 

What am I missing?

**Comment by mikefarah** [comment, marked as answer]
Yeah the bash completion script is just for commands and flags - it's generated from the CLI library tool I'm using which is why it might be so long.
